### PR TITLE
Update `README.md` to reflect new modernization checks

### PR DESCRIPTION
**Pull request by alvrogd**
In recent updates, including #14, #15, #16, #17, #18, #20, several new modernization checks were introduced to the Open Catalog. While some modernization checks have been part of the Catalog for some time, their importance has significantly
increased recently.

Consequently, we're updating the repository description to highlight two primary areas of focus: modernization and optimization.

Additionally, I've taken the opportunity to enhance a bit the readability of the existing text.

### CODE REVIEW

**Review by inaki-amatria — APPROVED** [review]
Looks great!

File: README.md [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 
-This catalog is a collaborative effort to consolidate the collective wisdom of
-performance experts on the best practices for performance. It consist of a
-glossary and a list of checks for the C, C++ and Fortran programming languages.
+This Open Catalog is a collaborative effort to consolidate expert knowledge on
+best practices for modernizing and optimizing code written in C, C++, and
+Fortran programming languages. The Catalog consists of a comprehensive set of
+checks (rules) that describe specific issues in the source code and provide
+guidance on corrective actions, along with extensive documentation, example
+codes, and references to additional reading resources.
```
Comment (jgonzac) [inline README.md:10]:
```suggestion
codes and references to additional reading resources.
```

File: README.md [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
 
 ## Benchmarks
 
-This catalog includes [a suite of microbenchmarks](Benchmark/README.md) that
-demonstrate the potential performance improvement that can be attained by
-applying the fixes that are suggested for each check.
+The Catalog includes [a suite of microbenchmarks](Benchmark/README.md) designed
```
Comment (jgonzac) [inline README.md:14]: Maybe?
```suggestion
The Open Catalog includes [a suite of microbenchmarks](Benchmark/README.md) designed
```

**Review by jgonzac — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (alvrogd) [reply README.md]: Done!

File: README.md (hunk 1 above)
Comment (alvrogd) [reply README.md]: Done!

**Review by ruifm — APPROVED** [review]

**Pull request merged**